### ISSUE TITLE
Unexpected interaction between --allowed_hosts and enable_socket()

### ISSUE THREAD

**Issue by okomarov** [miketheman/pytest-socket#27]
I have the following general setting:
```
[pytest]
addopts = --allow-hosts=127.0.0.1,::1
```

and `socket_enabled()` (or passing the fixture) to a specific test errors at the connection level. The _true_socket is restored, but the connection is not:
```
inst = <socket.socket fd=21, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('0.0.0.0', 0)>, args = (('***.***.42.194', 443),), host = '***.***.42.194'

    def guarded_connect(inst, *args):
        host = host_from_connect_args(args)
        if host and host in allowed:
            return _true_connect(inst, *args)
>       raise SocketConnectBlockedError(allowed, host)
E       Failed to send request: A test tried to use socket.socket.connect() with host "***.***.42.194" (allowed: "127.0.0.1,::1").
```

If I am explicitly enabling the socket for a test I expected it to override the `allowed_hosts`.

Adding `remove_host_restrictions()` inside the specific test restores the `_true_connect` and allows the test to run.

Maybe `enable_socket` should be:
```
def enable_socket():
    """ re-enable socket.socket to enable the Internet. useful in testing.
    """
    socket.socket = _true_socket
    socket.socket.connect = _true_connect
```

Notice the `socket.socket.connect = _true_connect`. This is also probably what happens with issue https://github.com/miketheman/pytest-socket/issues/15

**Comment by miketheman** [comment]
Thanks for reporting in detail! I haven't spent much time on this project as of late, but will try to evaluate this (and other issues) in a few weeks.

**Comment by galonsky** [comment]
I'm experiencing this issue too - using the fixture works, but it would be nice to use the mark reliably too.

**Comment by miketheman** [comment]
In #42 I added a fix that should have fixed this, and was released on 0.3.5.
Has that been resolved? Going to close this issue, please reopen with further details if it's still an issue!
